### ISSUE TITLE
don't allow slashes in resource names

### ISSUE THREAD

**Issue by sosiouxme** [openshift/origin#1844]
Create an imagestream with following content:

```
    {
      "apiVersion": "v1beta1",
      "kind": "ImageStream",
      "metadata": {
        "name": "openshift/hello-openshift"
      }
    }
```

Now try to delete it. (This is with user "joe" in project "sinatra" but that doesn't matter much.)

```
$ osc get imageStreams                                                                                                                                                                           
NAME                        DOCKER REPO   TAGS
openshift/hello-openshift                 
$ osc delete imageStreams openshift/hello-openshift                                                                                    
Error: when passing arguments in resource/name form, all arguments must include the resource
$ osc delete imageStreams hello-openshift
Error from server: imageStream "hello-openshift" not found
$ osc delete imageStreams/openshift/hello-openshift
Error: arguments in resource/name form may not have more than one slash
$ osc delete imageStreams --all
Error: "/osapi/v1beta1/imageStreams/openshift/hello-openshift?namespace=sinatra" is forbidden because joe cannot delete on imageStreams/hello-openshift with name "openshift" in sinatra
```

1. the resource parser is confused by the slash in the name
2. the API call to delete is also confounded by the slash in the name

@deads2k et al

**Comment by sosiouxme** [comment]
Extra note, you can't even delete the project to get rid of the thing once it's created.

**Comment by liggitt** [comment]
fixed in #1847
